File: app/src/main/java/com/urban/mvvmshowcase/view/adapter/StandardOneLineViewHolder.java
package com.urban.mvvmshowcase.view.adapter;


import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.urban.mvvmshowcase.model.entity.Person;

public class StandardOneLineViewHolder extends RecyclerView.ViewHolder implements View.OnClickListener {
    private final ClickListener clickListener;
    private final TextView textLine;
    private Person person;

    StandardOneLineViewHolder(@NonNull ViewGroup container,
                              @NonNull ClickListener clickListener) {
        super(LayoutInflater.from(container.getContext())
                .inflate(android.R.layout.simple_list_item_1, container, false));
        this.clickListener = clickListener;
        this.textLine = (TextView) itemView.findViewById(android.R.id.text1);

        itemView.setOnClickListener(this);
    }

    void bind(Person person) {
        textLine.setText(person.toString());
        this.person = person;
    }

    @Override
    public void onClick(View v) {
        clickListener.onPersonClick(person);
    }
}

File: app/src/main/java/com/urban/mvvmshowcase/person/list/adapter/StandardOneLineViewHolder.kt
package com.urban.mvvmshowcase.person.list.adapter

import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import com.urban.mvvmshowcase.person.Person

class StandardOneLineViewHolder(container: ViewGroup, private val clickListener: ClickListener) :
        RecyclerView.ViewHolder(LayoutInflater.from(container.context)
                .inflate(android.R.layout.simple_list_item_1, container, false)),
        View.OnClickListener {
    private lateinit var person: Person
    private val textLine = itemView.findViewById(android.R.id.text1) as TextView

    init {
        itemView.setOnClickListener(this)
    }

    fun bind(person: Person) {
        textLine.text = person.toString()
        this.person = person
    }

    override fun onClick(v: View?) = clickListener.onPersonClick(person)
}